libkbfs/tlf_handle.go

```diff
@@ -552,6 +592,56 @@ func (h *TlfHandle) ResolveAgain(ctx context.Context, resolver resolver) (
 	return h.ResolveAgainForUser(ctx, resolver, keybase1.UID(""))
 }
 
+type partialResolver struct {
+	unresolvedAssertions map[string]bool
+	delegate             resolver
+}
+
+func (pr partialResolver) Resolve(ctx context.Context, assertion string) (
+	libkb.NormalizedUsername, keybase1.UID, error) {
+	if pr.unresolvedAssertions[assertion] {
+		// Force an unresolved assertion.
+		return libkb.NormalizedUsername(""),
+			keybase1.UID(""), NoSuchUserError{assertion}
+	}
+	return pr.delegate.Resolve(ctx, assertion)
+}
+
+// ResolvesTo returns whether this handle resolves to the given one.
+// It also returns the partially-resolved version of h; this should
```

Comment:
Should define what "partially-resolved" means -- it doesn't attempt to resolve any assertions that are also unresolved in `other`.